### DISCUSSION TITLE
Probleme mit tsc

### DISCUSSION THREAD

**Discussion by 0100111101111010** [VDI-CodING/codeING-main#25] [Q&A]
tsc --init führt zu dem Problem (siehe unten):

![grafik](https://user-images.githubusercontent.com/76489487/111512185-c6a80b80-874f-11eb-8920-e071d3d087a3.png)

**Comment by stan1025** [comment]
Hast du die Powershell mit Administrationsrechten geöffnet? Die PowerShell braucht zur Ausführung vieler Script-Funktionen Admin-Rechte.

**Comment by stan1025** [comment]
Sollte das Problem nach dem Start mit Admin-Rechten weiterhin bestehen, nutzt den Befehl:
npx tsc --init

**Reply by stan1025** [reply]
Gilt auch für Linux

**Reply by KlausLex** [reply]
ebenfalls: npx touch src/app.ts :)

**Comment by workstation24-7** [comment]
<img width="717" alt="Unbenannt" src="https://user-images.githubusercontent.com/50848429/111516254-fd802080-8753-11eb-90d7-256c1baba75b.PNG">

**Reply by stan1025** [reply]
Hi @workstation24-7 ,
ich würde dir bzw. allen Raten, den Default für das Terminal in Visual Studio Code auf die Kommandozeile zu ändern und nicht auf Powershell zu lassen. Die PowerShell ist das deutlich mächtigere Tool und kann deutlich mehr "Schaden" auf eurem Rechner anrichten - was sich darin äußert, dass es mehr Restriktionen in der Nutzung gibt. 

Ich habe hier mal ein Screenshot meines VS-Code Fensters reinkopiert und markiert, wo du / ihr klicken müsst, um das Default-Terminal zu ändern.

![image](https://user-images.githubusercontent.com/9693984/111884436-ee9ea580-89c1-11eb-8b38-3e400061343e.png)

Im Anschluss öffnet sich ein Fenster oben in der Mitte mit Auswahlmöglichkeiten.
Hier bitte 'Command Prompt' anklicken.

![image](https://user-images.githubusercontent.com/9693984/111884498-463d1100-89c2-11eb-8a27-05c3ed608f25.png)

Von nun an, wird das Terminal immer als Kommandozeile von Windows gestartet.
Da sollten die CmdLet Probleme nicht mehr auftreten und die Befehle können wie beschrieben direkt eingegeben werden.

Falls das auch nicht hilft bitte nochmal melden!

**Reply by su-ro** [reply]
Das Dojo S1D0 konnte ich ausführen. Leider kompiliert meine Übungsdatei nicht. Ich habe alles ausprobiert, was ihr vorgeschlagen habt (PowerShell als Admin ausgeführt,  npx tsc --init, npx touch). 
Nachdem ich die Default Shell auf cmd geändert habe, kommt "Der Befehl "tsc" ist entweder falsch geschrieben oder konnte nicht gefunden werden." 
Habt ihr noch Ideen?

![grafik](https://user-images.githubusercontent.com/65827158/112056816-c301f380-8b58-11eb-9f29-9ba7670283d2.png)

![grafik](https://user-images.githubusercontent.com/65827158/112056876-dad97780-8b58-11eb-9155-ce2a5be976d2.png)

**Reply by Blumi102** [reply]
Hallo @su-ro :)

Für die ersten beiden Dojos genügt es, wenn du deinen Code in die app.ts reinschreibst, die wir im Zuge der Einrichtung angelegt haben. Diese kannst du dann mit ```npm run start-watch``` starten.
Die Erstellung neuer .ts-Datein und wie man diese einbindet, wird dann in S1D3 gezeigt.

Viele Grüße,
Michelle
